pipelining_example/pipelining_example.c

```diff
@@ -0,0 +1,2793 @@
+#define _GNU_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#include <arpa/inet.h>
+#include <byteswap.h>
+#include <endian.h>
+#include <getopt.h>
+#include <infiniband/verbs.h>
+#include <infiniband/mlx5dv.h>
+#include <inttypes.h>
+#include <netdb.h>
+#include <signal.h>
+#include <time.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#ifdef __x86_64__
+#define CACHE_LINE_SIZE 64
+#elif defined(__aarch64__)
+#define CACHE_LINE_SIZE 128
+#else
+#error Unknown architecture
+#endif
+#define CACHE_LINE_MASK (CACHE_LINE_SIZE - 1)
+#define CACHE_LINE_ALIGN(_x) ((_x + CACHE_LINE_MASK) & ~CACHE_LINE_MASK)
```

Comment:
This is not used, please remove it.

Reply:
OK, I will remote it, thanks!

Reply:
I found it's used for allocating recv_buf/send_buf, so we should keep it.

Reply:
This is a performance optimization. Using such performance optimisation in code examples is redundant and confusing for me. Because it seems like the API cannot work with non cache aligned buffers.

I would drop the buffer alignment because it is not related to the feature showed in the example. What do you thinkt?

Reply:
Ok, I will drop it.